pkgs/by-name/mu/muse-sounds-manager/package.nix

```
@@ -0,0 +1,69 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  autoPatchelfHook,
+  dpkg,
+  fontconfig,
+  zlib,
+  icu,
+  libX11,
+  libXext,
+  libXi,
+  libXrandr,
+  libICE,
+  libSM,
+  openssl,
+}:
+
+stdenv.mkDerivation rec {
```

Should speed up eval time:

```suggestion
stdenv.mkDerivation (finalAttrs: {
```

---

I have studied interpreters and compilers, and I am sure that changing a simple lexical lookup to a lookup through many levels of indirection that happen to the lexical set of attributes before they become `finalAttrs` can not speed anything up. I do not expect that it would noticeably slow evaluation of any attributes of this derivation down either, since attribute lookup should not be a significant factor anyway.

However, even if this had an impact on performance, it is still a stylistic choice and should be a matter of policy or consensus, not a matter of performance.

---

Yes, this should not have a positive impact on eval speed?

---

Yea you are right @orivej that using a function instead of `rec` might not speed eval time in this case. However, there is I think a consensus that avoiding `rec` is recommended:

https://nix.dev/guides/best-practices#recursive-attribute-set-rec